core/schema/module.go

```diff
@@ -37,26 +37,23 @@ func (s *moduleSchema) Install() {
 			ArgDoc("line", "The line number within the filename.").
 			ArgDoc("column", "The column number within the line."),
 
-		dagql.FuncWithCacheKey("currentModule", s.currentModule, core.CachePerClient).
+		dagql.FuncWithCacheKey("currentModule", s.currentModule, dagql.CachePerClient).
 			Doc(`The module currently being served in the session, if any.`),
 
-		dagql.Func("currentTypeDefs", s.currentTypeDefs).
-			// Impure for now, could use a finer grain cache key if we had the ability to mix
-			// a digest of the dagql server schema into the cache key.
-			Impure("Can change when modules are loaded into the schema.").
+		dagql.FuncWithCacheKey("currentTypeDefs", s.currentTypeDefs, dagql.CachePerCall).
```

Comment:
neat - though, not sure about the name `CachePerCall` 🤔 - initially it sounded like the opposite of what it does (the call itself isn't cached since you get a different result each time, but each result is cached)

that said, don't know what else to suggest.

Reply:
Yeah it's a very subtle thing that's hard to summarize in a short name... Nothing better in terms of names is coming to mind for me either, but if I think of anything before merge I'll update it.